lib/transformers/typescript.js

```diff
@@ -16,6 +52,21 @@ module.exports = {
     const tsconfig = getTsJestConfig(config)
     const babelOptions = getBabelOptions(filePath)
 
+    if (vueJestConfig.enableExperimentalTsDiagnostics) {
+      const diagnostics = getDiagnostics(
+        scriptContent,
+        { ...tsconfig, noEmitOnError: true },
```

Comment:
`ts-jest` config always:

-  set `sourceMap` to [true](https://github.com/kulshekhar/ts-jest/blob/452639242d7f59502bda7aa94b81735f9c9ad203/src/config/config-set.ts#L590). Since you don't need `sourceMap` here, you can try setting it to `false` to TypeScript `Program` won't try to emit source map -> improve performance. 

- set `noEmit` to [false](https://github.com/kulshekhar/ts-jest/blob/452639242d7f59502bda7aa94b81735f9c9ad203/src/config/config-set.ts#L595). Probably here you can try setting it to `true` so `program.emit()` won't try to emit any files. -> improve performance.

When using `program.emit()`, TypeScript will also try to emit files, which can include compiled js, source map and type declaration.